offlineimap/imapserver.py

```diff
@@ -198,7 +198,7 @@ def __md5handler(self, response):
         challenge = response.strip()
         self.ui.debug('imap', '__md5handler: got challenge %s'% challenge)
 
-        passwd = self.__getpassword()
+        passwd = self.__getpassword().encode('ascii')
```

Comment:
I think I managed to find a compromise.  The encoding issues should be all resolved, yet this still won't work on Python 3.4 due to [Python bug](https://bugs.python.org/issue18240).  To make this work with older version we can consider few options :

1. Create our own `bytes()` that would accept encoding type as a second paramater.
2. Discover Python version and use bytes when py3

What do you think?

```diff
diff --git a/offlineimap/imapserver.py b/offlineimap/imapserver.py
index f373326..bea7017 100644
--- a/offlineimap/imapserver.py
+++ b/offlineimap/imapserver.py
@@ -199,7 +199,8 @@ class IMAPServer(object):
         self.ui.debug('imap', '__md5handler: got challenge %s'% challenge)
 
         passwd = self.__getpassword()
-        retval = self.username + ' ' + hmac.new(passwd, challenge).hexdigest()
+        retval = self.username + ' ' + hmac.new(bytearray(passwd, 'utf-8'),
+                                                bytearray(challenge, 'utf-8')).hexdigest()
         self.ui.debug('imap', '__md5handler: returning %s'% retval)
         return retval
```